consensus/istanbul/backend/engine.go

```diff
@@ -272,29 +270,103 @@ func (sb *Backend) verifyCommittedSeals(chain consensus.ChainReader, header *typ
 		return errEmptyCommittedSeals
 	}
 
+	// Check the signatures on the current header
+	snap, err := sb.snapshot(chain, header.Number.Uint64()-1, header.ParentHash, parents)
+	if err != nil {
+		return err
+	}
 	validators := snap.ValSet.Copy()
-	// Check whether the committed seals are generated by parent's validators
+	err = sb.checkValidatorSignatures(header.Hash(), validators, extra.Bitmap, extra.CommittedSeal)
+	if err != nil {
+		return err
+	}
+
+	// Check parent signatures only after block 1.
+	// The genesis block is skipped since it has no parents
+	// The first block is also skipped since its parent
+	// is the genesis block which contains no parent signatures
+	if number > 1 {
+		sb.logger.Debug("verifyCommittedSeals: verifying parent block for block", "num", number)
+		var parentValidators istanbul.ValidatorSet
+		// parents of epoch blocks have different validator sets since they
+		// belong to the previous epoch.
+		if number%sb.config.Epoch == 0 {
```

Comment:
We actually want to check if this is 1. `number%sb.config.Epoch == 0` means `number` is the last block of the epoch.